apps/src/JSInterpreter.js

```diff
@@ -563,6 +579,44 @@ JSInterpreter.prototype.getNearestUserCodeLine = function () {
 };
 
 /**
+ * Creates a property in the interpreter's global scope. When a parent is
+ * supplied and that parent object is in codegen's customMarshalObjectList,
+ * property gets/sets in the interpreter will be reflected on the native parent
+ * object. Functions can also be inserted into the global namespace using this
+ * method. If a parent is supplied, they will be invoked natively with that
+ * parent as the this parameter.
+ *
+ * @param {String} name Name for the property in the global scope.
+ * @param {*} value Native value that will be marshalled to the interpreter.
+ * @param {Object} parent (Optional) parent for the native value.
+ */
+JSInterpreter.prototype.createGlobalProperty = function (name, value, parent) {
```

Comment:
This is an exciting capability.  I wonder if we'll eventually want to expose this to levelbuilders somehow.